distribution/sc-proxy-dex/src/proxy_farm.rs

```diff
@@ -348,15 +318,94 @@ pub trait ProxyFarmModule: proxy_common::ProxyCommonModule + proxy_pair::ProxyPa
         farm_token_id: &TokenIdentifier,
         farm_token_nonce: Nonce,
         amount: &Self::BigUint,
-    ) -> ExitFarmResultType<Self::BigUint> {
-        self.farm_contract_proxy(farm_address.clone())
-            .exitFarm(
-                farm_token_id.clone(),
-                amount.clone(),
-                farm_token_nonce,
-                OptionalArg::Some(BoxedBytes::from(ACCEPT_PAY_FUNC_NAME)),
+    ) -> SCResult<ExitFarmResultType<Self::BigUint>> {
+        let mut arg_buffer = ArgBuffer::new();
+        arg_buffer.push_argument_bytes(BoxedBytes::from(ACCEPT_PAY_FUNC_NAME).as_slice());
+
+        let result = self.send().direct_esdt_nft_execute(
+            farm_address,
+            farm_token_id,
+            farm_token_nonce,
+            amount,
+            self.blockchain().get_gas_left(),
+            &b"exitFarm"[..],
+            &arg_buffer,
+        );
+        require!(result == Result::Ok(()), "Failed to call transfer execute");
+
+        let accented_funds_len = self.current_tx_accepted_funds().len();
+        require!(
+            accented_funds_len == 1 || accented_funds_len == 2,
+            "Bad received funds len"
+        );
+
+        let asset_token_id = self.asset_token_id().get();
+        let locked_asset_token_id = self.locked_asset_token_id().get();
+        let mut farming_token = Option::<FftTokenAmountPair<Self::BigUint>>::None;
+        let mut reward_token = Option::<GenericEsdtAmountPair<Self::BigUint>>::None;
+
+        for (token_id, token_nonce) in self.current_tx_accepted_funds().keys() {
```

Comment:
wouldn't be better to use the validate function ? it looks relatively the same.

Reply:
like lines 342-385 could be taken out into a validate function

Reply:
done